### DISCUSSION TITLE
Docker Desktop Installer (from the script) licensing concern

### DISCUSSION THREAD

**Discussion by pranchan007** [sakai135/wsl-vpnkit#65] [Q&A] [answered]
As we see in the script that it downloads docker desktop installer temporarily and extract vpnkit out of it and uses it. Can this be a licensing concern for the machines which do not want to use docker desktop paid subscriptions?

I have managed to install docker on Windows on top of WSL 2 + Ubuntu without Docker Desktop (as it is generally installed on any Ubuntu machine), however I have noticed that sometimes the internet goes away either in the WSL and/or in docker containers. I believe, vpnkit addresses this problem (if no, please help me to understand).

I thought to use this library but saw in the script that, it downloads the docker desktop installer temporarily, so just wanted to check if this violates T&C for docker desktop (for situation like I'm in where I do not want to opt for Docker Desktop due to paid subscription)?

Or can't vpnkit be take directly from [here](https://github.com/moby/vpnkit) in the script?

Please help to understand this?

**Comment by sakai135** [comment, marked as answer]
@pranchan007 

If your connectivity goes away unrelated to connecting your Windows host to a VPN or any network, I am unsure if this will solve your issue. I would check if you can ping IP addresses but not hostnames, and if that's the case, just setting your own DNS servers will probably fix the issue.

I am not really an expert on licensing, but strictly speaking, it could be a violation. There is supposedly a grace period until the end of January 2022, so I intend to replace the VPNKit binaries to ones built from the VPNKit repo before then. The VPNKit repo currently does not build the Windows binary as part of their automated process, and they do not have any pre-built binaries available for download (https://github.com/moby/vpnkit/pull/544).

**Comment by pranchan007** [comment]
@sakai135 
Thanks for the prompt response and your support on the queries raised.

I think my connectivity goes away irrespective of VPN. I recently tested with connecting/disconnecting to VPN and in each scenario everything worked (local connection as well as internet). The behaviour that I have been observing that after few days of using my setup, suddenly internet in WSL and docker container goes away. I have also noticed that if I reset WSL e.g. enable/disable WSL feature or any such config which refreshes WSL n/w, the connection is restored back. I'm pretty sure after few days my working setup will again face same issue but what is causing this is not being figured out. I recently ended up finding this library and thought that maybe automatic disconnection of VPN might be changing some IP settings (like it does in n/w adapter) and this library may help me to keep it consistent.

Also, when the n/w connection is interrupted for WSL itself and/or the docker container, it is not able to ping to public servers whereas internal IPs work, similarly hostnames also work.

Now, I'm little confused if everything is working irrespective of VPN and also this library is not to solve internet issues in WSL or container, then when do we exactly need this :) I read the documentation but still not much clear.

NOTE: Setting up a custom DNS has never helped me. However, I have observed a behaviour that, in WSL 1, it sets up WSL dns from host as well as vpn's dns in resolv.conf whereas in WSL 2, it just sets up one for WSL from host.

And thanks for the clarification over licensing stuff and I'm very happy to see that you are looking to replace the binary before the grace period. That would help many people. Keep up the good work!

**Reply by sakai135** [reply]
wsl-vpnkit is made to restore network connectivity where the VPN blocks network connectivity from WSL 2 when connected. I'll clarify that in a future update.

**Reply by pranchan007** [reply]
@sakai135 , I recently observed that you have removed the dependency from docker desktop installer and hosted your own copy. Happy to see that. Appreciate your all hard work. Also, I got an instance where WSL faced internet issue with VPN but was working w/o VPN. So I got chance to try out this wsl-vpnkit and it worked great! Thanks again!
